lineapy/transformer/node_transformer.py

```diff
@@ -278,6 +279,30 @@ def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
             attributes=create_lib_attributes(node.names),
         )
 
+    def visit_If(self, node: ast.If) -> Any:
+        test_call_node = self.visit(node.test)
+
+        if self.tracer.executor._id_to_value[test_call_node.id]:
+            exec = node.body
+            unexec = node.orelse
+        else:
+            exec = node.orelse
+            unexec = node.body
+        else_source = self.get_else_source(node)
+        self.tracer.control_node(
+            IfElseNode,
+            test_call_node,
+            self.get_source(node.test),
+            self.get_black_box_without_executing(unexec)
+            if else_source is not None
+            else None,
```

Comment:
i'm worried about the case where there is an if block with no else and a false test condition. in that case currently we include the entire if as a blackbox but this will skip the whole if. add a test case for this.

Reply:
I dont think we include the if condition.

Example:
```
a = [1,2,3]

if len(a) > 4:
  a.append(2)
```

Save artifact on `a`